src/test/java/com/wikia/webdriver/common/logging/Log.java

```diff
@@ -0,0 +1,338 @@
+      }
+      Log.logJSError();
+    }
+  }
+
+  public static void stacktrace(Throwable throwable) {
+    String exceptionMessage = Throwables.getStackTrace(throwable);
+    List<LogData> classList = new ArrayList<>();
+    classList.add(LogLevel.ERROR);
+    classList.add(LogType.STACKTRACE);
+
+    VelocityWrapper
+        .fillLogRow(classList, "STACKTRACE", escapeHtml(exceptionMessage).replace("\n", "<br>"));
+  }
+
+  public static void stop() {
+    WikiaWebDriver driver = DriverProvider.getActiveDriver();
+    if (driver.getProxy() != null && Configuration.getForceHttps()) {
+      Har har = driver.getProxy().getHar();
+      for (HarEntry entry : har.getLog().getEntries()) {
+        URL url;
+        try {
+          url = new URL(entry.getRequest().getUrl());
+          if (url.getHost().contains("wikia")) {
+            boolean isHttps = entry.getRequest().getUrl().startsWith("https");
+            Log.log("VISITED URL", "Url: " + entry.getRequest().getUrl(),
+                    !Configuration.getForceHttps() || isHttps);
+          }
+        } catch (MalformedURLException e) {
+          Log.log("MALFORMED URL", "Url: " + entry.getRequest().getUrl(), false);
+        }
+      }
+    }
+    String html = VelocityWrapper.fillLastLogRow();
+    CommonUtils.appendTextToFile(Log.LOG_PATH, html);
+    Log.testStarted = false;
+  }
+
+  public static void startReport() {
+    CommonUtils.createDirectory(Log.SCREEN_DIR_PATH);
+    imageCounter = 0;
+
+    String date = DateTimeFormat.forPattern(Log.DATE_FORMAT).print(DateTime.now(DateTimeZone.UTC));
+    String
+        polishDate =
+        DateTimeFormat.forPattern(Log.POLISH_DATE_FORMAT).print(DateTime.now().withZone(DateTimeZone
+                                                                                            .forTimeZone(
+                                                                                                TimeZone
+                                                                                                    .getTimeZone(
+                                                                                                        "Europe/Warsaw"))));
+    String browser = Configuration.getBrowser();
+    String os = System.getProperty("os.name");
+    String testingEnvironmentUrl = UrlBuilder.createUrlBuilder().getUrl();
+    String testingEnvironment = Configuration.getEnv();
+    String testedVersion = "TO DO: GET WIKI VERSION HERE";
+
+    String
+        headerHtml =
+        VelocityWrapper.fillHeader(date, polishDate, browser, os, testingEnvironmentUrl,
+                                   testingEnvironment, testedVersion);
+    CommonUtils.appendTextToFile(Log.LOG_PATH, headerHtml);
+    appendShowHideButtons();
+    try {
+      FileInputStream input = new FileInputStream("./src/test/resources/script.txt");
+      String content = IOUtils.toString(input);
+      CommonUtils.appendTextToFile(Log.LOG_PATH, content);
+    } catch (IOException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)